ui_test/POM_hiway_asg/pages/timesheet.py

```diff
@@ -0,0 +1,156 @@
+            return True
+        else:
+            return False
+
+    def verify_total_work_time(self):
+        timesheet_entries = self.driver.find_elements_by_name(TIMESHEET_HOUR_FIELD_NAME)
+        hour_total = 0
+        min_total = 0
+        count = 0
+        for timesheet_entry in timesheet_entries:
+            min_end_text = ',"project_code"'
+            if count > 0:
+                min_end_text = ',"description'
+            count = 1
+            entry = timesheet_entry.get_attribute(TIMESHEET_ENTRY_ATTRIBUTE)
+            hour_start = entry.find('"hrs":') + 6
+            hour_end = entry.find(',"min":')
+            hour_total += int(entry[hour_start:hour_end])
+            min_start = entry.find('"min":') + 6
+            min_end = entry.find(min_end_text)
+            min_total += int(entry[min_start:min_end])
+
+        # adding all the time of the timesheet page and converting it to minutes
+
+        total_mins = min_total + hour_total * 60
+        timesheet_time = self.read_timesheet_time()
+        timesheet_hour = int(timesheet_time[:2])
+        timesheet_min = int(timesheet_time[3:5])
+        timesheet_total_mins = timesheet_hour * 60 + timesheet_min
+        if timesheet_total_mins == total_mins:
+            return total_mins, True
+        return total_mins, False
+
+    def status_of_next_btn_on_today_date(self):
+        driver = self.driver
+        if driver.find_element_by_xpath(TIMESHEET_NEXT_BTN_XPATH).is_enabled():
+            return True
+        return False
+
+    def verify_previous_btn_click(self, click_count):
+        driver = self.driver
+        for click in range(click_count):
+            driver.find_element_by_xpath(TIMESHEET_PREVIOUS_BTN_XPATH).click()
+        timesheet_time = self.read_timesheet_time()
+        timesheet_date = timesheet_time[-6:]
+        prev_date = date.today() - timedelta(click_count)
+        prev_date = prev_date.strftime('%b %d')
+        if prev_date == timesheet_date:
+            return True
+        return False
+
+    def verify_task_entry_adding(self, task_code, task_type, task_hour, task_min, task_dsc):
+        driver = self.driver
+        self.make_timesheet_entry(task_code, task_type, task_hour, task_min, task_dsc)
+        task_data = driver.find_element_by_name(TIMESHEET_HOUR_FIELD_NAME).get_attribute(TIMESHEET_ENTRY_ATTRIBUTE)
+        get_task_code = task_data[task_data.find('code":"') + 7:task_data.find('","type')]
+        get_task_type = task_data[task_data.find('type":"') + 7:task_data.find('","description')]
+        get_task_hour = task_data[task_data.find('"hrs":') + 6:task_data.find(',"min"')]
+        get_task_min = task_data[task_data.find('"min":') + 6:task_data.find(',"project')]
+        get_task_dsc = task_data[task_data.find('description":"') + 14:task_data.find('","state"')]
+        if task_code == get_task_code and task_type == get_task_type and task_hour == get_task_hour and task_min == get_task_min and task_dsc == get_task_dsc:
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The line contains 158 characters which is greater than 120 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=python%3ALineLength)